beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -141,17 +179,14 @@ func (s *Store) insert(ctx context.Context,
 			s.treeRootNode = n
 			s.headNode = n
 			s.highestReceivedNode = n
+			s.proposerHeadNode = n
 		} else {
 			return n, errInvalidParentRoot
 		}
 	} else {
 		parent.children = append(parent.children, n)
 		// Apply proposer boost
-		timeNow := uint64(time.Now().Unix())
-		if timeNow < s.genesisTime {
-			return n, nil
-		}
-		secondsIntoSlot := (timeNow - s.genesisTime) % params.BeaconConfig().SecondsPerSlot
+		secondsIntoSlot := (timestamp - s.genesisTime) % params.BeaconConfig().SecondsPerSlot
```

Comment:
duplicated logic

Reply:
I didn't understand this one

Reply:
ah i mean that i've seen a few lines of code in the PR that compute seconds into a slot that could just a single helper function